File: java/src/main/java/com/carepet/model/PetDAO.java
package com.carepet.model;

import com.datastax.oss.driver.api.mapper.annotations.Dao;
import com.datastax.oss.driver.api.mapper.annotations.Insert;
import com.datastax.oss.driver.api.mapper.annotations.Select;
import com.datastax.oss.driver.api.mapper.annotations.Update;

import java.util.UUID;

@Dao
public interface PetDAO {
    @Insert
    void create(Pet pet);

    @Update
    void update(Pet pet);

    @Select
    Pet get(UUID owner, UUID id);
}

File: csharp/Model/PetDAO.cs
using System;
using System.Collections.Generic;
using Cassandra;
using Cassandra.Mapping;

namespace CarePet.Model
{
    public class PetDAO
    {
        private readonly IMapper _mapper;

        public PetDAO(ISession session)
        {
            _mapper = new Cassandra.Mapping.Mapper(session);
        }

        public void Create(Pet pet)
        {
            _mapper.Insert(pet);
        }

        public void Update(Pet pet)
        {
            _mapper.Update(pet);
        }

        public Pet Get(Guid ownerId, Guid petId)
        {
            return _mapper.FirstOrDefault<Pet>(
                "WHERE owner_id = ? AND pet_id = ?",
                ownerId,
                petId
            );
        }

        public IEnumerable<Pet> FindByOwner(Guid ownerId)
        {
            return _mapper.Fetch<Pet>(
                "WHERE owner_id = ?",
                ownerId
            );
        }
    }
}
